shell/plugins/steve/getters.js

```diff
@@ -35,48 +41,79 @@ export default {
     const parsedUrl = parse(url);
     const isSteve = steveRegEx.test(parsedUrl.path);
 
-    // labelSelector
-    if ( opt.labelSelector ) {
-      url += `${ url.includes('?') ? '&' : '?' }labelSelector=${ opt.labelSelector }`;
-    }
-    // End: labelSelector
+    const stevePagination = stevePaginationUtils.checkAndCreateParam(opt);
 
-    // Filter
-    if ( opt.filter ) {
-      url += `${ (url.includes('?') ? '&' : '?') }`;
-      const keys = Object.keys(opt.filter);
+    if (stevePagination) {
+      url += `${ (url.includes('?') ? '&' : '?') + stevePagination }`;
+    } else {
+      // labelSelector
+      if ( opt.labelSelector ) {
+        url += `${ url.includes('?') ? '&' : '?' }labelSelector=${ opt.labelSelector }`;
+      }
+      // End: labelSelector
 
-      keys.forEach((key) => {
-        let vals = opt.filter[key];
+      // Filter
+      if ( opt.filter ) {
+        url += `${ (url.includes('?') ? '&' : '?') }`;
+        const keys = Object.keys(opt.filter);
 
-        if ( !isArray(vals) ) {
-          vals = [vals];
-        }
+        keys.forEach((key) => {
+          let vals = opt.filter[key];
 
-        // Steve's filter options now support more complex filtering not yet implemented here #9341
-        if (isSteve) {
-          url += `${ (url.includes('filter=') ? '&' : 'filter=') }`;
-        }
+          if ( !isArray(vals) ) {
+            vals = [vals];
+          }
 
-        const filterStrings = vals.map((val) => {
-          return `${ encodeURI(key) }=${ encodeURI(val) }`;
+          // Steve's filter options now support more complex filtering not yet implemented here #9341
+          if (isSteve) {
+            url += `${ (url.includes('filter=') ? '&' : 'filter=') }`;
```

Comment:
It looks like alot of these changes are really just nesting so you probably don't want to touch them but in case you did the url manipulation in this file looks like it should either use stuff or move stuff to the utils/url module.

Reply:
yeah i cringed at this fn. the core change was just moving things around. i had thought about converting everything over to something like `<URL>.searchParams.append` but thought the PR was a big enough change as is.